src/index.js

```diff
@@ -75,7 +75,12 @@ HTTPSnippet.prototype.prepare = function (request) {
   if (request.headers && request.headers.length) {
     // loweCase header keys
     request.headersObj = request.headers.reduce(function (headers, header) {
-      headers[header.name.toLowerCase()] = header.value
+      var headerName = header.name
+      if (request.httpVersion.match(/^HTTP\/2/)) {
```

Comment:
As a performance bump, it would be good to declare this regex once and re-use it rather than constructing it within each iteration of `.reduce()` 😄